executor/join.go

```diff
@@ -533,6 +536,9 @@ func (e *HashJoinExec) buildHashTableForList() error {
 		valBuf  = make([]byte, 8)
 	)
 	for i := 0; i < e.innerResult.NumChunks(); i++ {
+		if e.finished.Load().(bool) {
```

Comment:
this check can be removed, because we only need to guarantee not calling the `next` function of the child.